drivers/vfs/driver.go

```diff
@@ -204,18 +203,33 @@ func (d *Driver) create(id, parent string, opts *graphdriver.CreateOpts, ro bool
 	return nil
 }
 
-func (d *Driver) dir(id string) string {
-	for i, home := range d.homes {
-		if i > 0 {
-			home = filepath.Join(home, d.String())
+func (d *Driver) dir2(id string, useImageStore bool) string {
+	var homedir string
+
+	if useImageStore && d.imageStore != "" {
+		homedir = d.imageStore
+	} else {
+		homedir = d.homes[0]
+	}
+	newpath := filepath.Join(homedir, d.String(), "dir", filepath.Base(id))
+	if _, err := os.Stat(newpath); err != nil {
```

Comment:
⚠️ How does this handle _any_ locking? The generic code maintains RW/RO locks on the individual images / layers. This… does nothing?

Reply:
this is still outstanding.

I don't see how this is different than the previous version, what locking do we need here?